libs/driver/in-memory/src/checkout/in-mem/checkout.testing.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, Inject } from '@angular/core';
+
+import { DaffCartFactory, DaffCartItemFactory, DaffProductImageFactory} from '@daffodil/core/testing';
+
+import { InMemoryDbService, STATUS } from 'angular-in-memory-web-api';
+import { DaffDriverInterface, DaffDriver } from '@daffodil/driver';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffInMemoryCheckoutTestingService implements InMemoryDbService {
+  private order;
+
+  constructor(
+    private cartFactory: DaffCartFactory,
+    private cartItemFactory: DaffCartItemFactory,
+    private productImageFactory: DaffProductImageFactory,
+    @Inject(DaffDriver) public driver: DaffDriverInterface
+  ) {
+    this.order = this.cartFactory.create({ 
+      items: this.cartItemFactory.createMany(2, {
+        image: this.productImageFactory.create()
+      }) 
+    });
+  }
+
+  post(reqInfo: any) {
+    return reqInfo.utils.createResponse$(() => {
+      if (reqInfo.id === 'placeOrder') {
+        //should make a service call to clear cart here.
```

Comment:
This is where we should do work to turn the cart into an order.

Reply:
I could actually build the order from the cart object, but I wouldn't have thought that the actual production call would post a `Cart`. Probably would just be a cart ID. And actually calling the in-memory service for the cart isn't straightforward as we discovered with the "clearCart" process. So I'm just creating a new `Order` from scratch.